geopandas/geodataframe.py

```diff
@@ -946,6 +947,14 @@ def dissolve(self, by=None, aggfunc="first", as_index=True):
         GeoDataFrame
         """
 
+        if by is None:
+            self["__dummy"] = True
+            by = "__dummy"
```

Comment:
I think doing something like `by = np.zeros(len(self), dtype="int64")` (so passing directly an array to group on), should also work and avoids adding a new column (adding new columns can sometimes have unforeseen costs, although not sure groupby will "consolidate" the dataframe first)

Reply:
I had no idea you can pass an array directly. Good point!

Reply:
I don't think there's any real performance implication either way, but I personally use `by = lamdba x: 0` for this, which feels more intuitive to me, but that's also very much a matter of perspective/familiarity.